vocode/streaming/agent/groq_agent.py

```diff
@@ -0,0 +1,223 @@
+            self.functions,
+            self.agent_config.prompt_preamble,
+        )
+
+        parameters: Dict[str, Any] = {
+            "messages": messages,
+            "max_tokens": self.agent_config.max_tokens,
+            "temperature": self.agent_config.temperature,
+            "model": self.agent_config.model_name,
+        }
+
+        if use_functions and self.functions:
+            parameters["functions"] = self.functions
+
+        return parameters
+    
+    async def _create_groq_stream(self, chat_parameters: Dict[str, Any]) -> AsyncGenerator:
+        try:
+            stream = await self.groq_client.chat.completions.create(**chat_parameters)
+        except Exception as e:
+            logger.error(
+                f"Error while hitting Groq with chat_parameters: {chat_parameters}",
+                exc_info=True,
+            )
+            raise e
+        return stream
+
+    def should_backchannel(self, human_input: str) -> bool:
+        return (
+            not self.is_first_response()
+            and not human_input.strip().endswith("?")
+            and random.random() < self.agent_config.backchannel_probability
+        )
+
+    def choose_backchannel(self) -> Optional[BotBackchannel]:
+        backchannel = None
+        if self.transcript is not None:
+            last_bot_message: Optional[Message] = None
+            for event_log in self.transcript.event_logs[::-1]:
+                if isinstance(event_log, Message) and event_log.sender == Sender.BOT:
+                    last_bot_message = event_log
+                    break
+            if last_bot_message and last_bot_message.text.strip().endswith("?"):
+                return BotBackchannel(text=self.post_question_bot_backchannel_randomizer())
+        return backchannel
+
+    async def generate_response(
+        self,
+        human_input: str,
+        conversation_id: str,
+        is_interrupt: bool = False,
+        bot_was_in_medias_res: bool = False,
+    ) -> AsyncGenerator[GeneratedResponse, None]:
+        assert self.transcript is not None
+
+        chat_parameters = {}
+        if self.agent_config.vector_db_config:
+            try:
+                docs_with_scores = await self.vector_db.similarity_search_with_score(
+                    self.transcript.get_last_user_message()[1]
+                )
+                docs_with_scores_str = "\n\n".join(
+                    [
+                        "Document: "
+                        + doc[0].metadata["source"]
+                        + f" (Confidence: {doc[1]})\n"
+                        + doc[0].lc_kwargs["page_content"].replace(r"\n", "\n")
+                        for doc in docs_with_scores
+                    ]
+                )
+                vector_db_result = (
+                    f"Found {len(docs_with_scores)} similar documents:\n{docs_with_scores_str}"
+                )
+                messages = format_openai_chat_messages_from_transcript(
```

Comment:
because we aren't token counting and truncating on the context window. We need an overwrite of this method for groq, as you had in your initial change. So let's use `format_groq_chat_messages_from_transcript()`